### PR TITLE
New insert tags system

### PR DESCRIPTION
**Pull request by ausi**
Followup to #4779 based on review from @m-vo

### CODE REVIEW

File: core-bundle/src/InsertTag/Flag/StringUtilFlag.php [hunk 1]
```diff
@@ -53,7 +53,8 @@ public function ampersand(InsertTagFlag $flag, InsertTagResult $result): InsertT
     {
         return $result
             ->withValue(StringUtil::ampersand($result->getValue()))
-            ->withOutputType(OutputType::text) // We can not safely assume HTML here as the method only encodes ampersands
+            // We can not safely assume HTML here as the method only encodes ampersands
```
Comment (leofeyer) [inline core-bundle/src/InsertTag/Flag/StringUtilFlag.php]:
```suggestion
            // We cannot safely assume HTML here as the method only encodes ampersands
```

File: core-bundle/src/InsertTag/InsertTagSubscription.php [hunk 2]
```diff
@@ -12,6 +12,9 @@
 
 namespace Contao\CoreBundle\InsertTag;
 
+/**
+ * @internal
+ */
```
Comment (leofeyer) [inline core-bundle/src/InsertTag/InsertTagSubscription.php:17]: Just a very minor nitpick: theoretically speaking, `@internal` already implies "do not extend", so `final` could be removed now.

Comment (ausi) [reply core-bundle/src/InsertTag/InsertTagSubscription.php:17]: But the `final` helps ourselfes to not inherit from it ☺️
I’m fine either way.

Comment (leofeyer) [reply core-bundle/src/InsertTag/InsertTagSubscription.php:17]: Me too, that‘s why I wrote "theoretically speaking". In my opinion, we should even use `final` more often (see #924).

**Review by m-vo — DISMISSED** [review]
Very nice! ♥️

LGTM from reading the code. I did not have time to manually test, though. 😊

File: core-bundle/src/DependencyInjection/Compiler/AddInsertTagsPass.php [hunk 3]
```diff
@@ -47,6 +49,7 @@ public function process(ContainerBuilder $container): void
                     }
 
                     $method = $this->getMethod($attributes['method'], $serviceTag, $container->findDefinition($serviceId)->getClass(), $serviceId);
+                    $attributes['resolveNestedTags'] ??= $this->getResolveNestedTagsFromMethod($container->findDefinition($serviceId)->getClass(), $method);
```
Comment (ausi) [inline core-bundle/src/DependencyInjection/Compiler/AddInsertTagsPass.php]:
```suggestion
                    $class = $container->findDefinition($serviceId)->getClass();
                    $method = $this->getMethod($attributes['method'], $serviceTag, $class, $serviceId);
                    $attributes['resolveNestedTags'] ??= $this->getResolveNestedTagsFromMethod($class, $method);
```

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**